shock2vr/src/save_load/held_item_save_data.rs

```
@@ -21,39 +36,110 @@ impl HeldItemSaveData {
         }
     }
 
-    // TODO: Refactor return value to result
     pub fn instantiate(
         &self,
         world: &mut World,
-    ) -> (Option<EntityId>, Option<EntityId>, Option<EntityId>) {
+    ) -> Result<(Option<EntityId>, Option<EntityId>, Option<EntityId>), HeldItemError> {
```

This doesn't actually seem any better. The point of switching to a `Result` is that this could be now be `Result<(EntityId, EntityId, EntityId), HeldItemError>` - removing the nested options. In other words, we should be able to remove all or nothing.

---

Thanks for pointing this out! The hands and inventory slots can legitimately be empty when we serialize, so we still need the outer Option to represent "nothing equipped". To make the Result meaningful, I added explicit remapping checks: if the save data claims an entity but we fail to remap it, we now bail with a slot-specific error instead of silently returning None. That keeps the valid empty-hand cases but catches the inconsistent ones.